csr_check_test.go

```diff
@@ -1004,36 +1156,906 @@ func TestAuthorizeCSR(t *testing.T) {
+					Spec: certificatesv1beta1.CertificateSigningRequestSpec{
+						Usages: []certificatesv1beta1.KeyUsage{
+							certificatesv1beta1.UsageKeyEncipherment,
+							certificatesv1beta1.UsageDigitalSignature,
+							certificatesv1beta1.UsageClientAuth,
+						},
+						Username: "system:serviceaccount:openshift-machine-config-operator:node-bootstrapper",
+						Groups: []string{
+							"system:authenticated",
+							"system:serviceaccounts:openshift-machine-config-operator",
+							"system:serviceaccounts",
+						},
+					},
+				},
+				csr: clientExtraO,
+			},
+			wantErr: "Doesn't match expected prefix",
+		},
+		{
+			name: "client with DNS",
+			args: args{
+				machines: []machinev1beta1.Machine{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"machine.openshift.io/cluster-api-machine-role": "worker",
+								"machine.openshift.io/cluster-api-machine-type": "worker",
+							},
+						},
+						Status: machinev1beta1.MachineStatus{
+							Addresses: []corev1.NodeAddress{
+								{
+									Type:    corev1.NodeInternalDNS,
+									Address: "monkey",
+								},
+							},
+						},
+					},
+				},
+				nodeName: "monkey",
+				nodeErr:  errors.NewNotFound(schema.GroupResource{}, ""),
+				req: &certificatesv1beta1.CertificateSigningRequest{
+					Spec: certificatesv1beta1.CertificateSigningRequestSpec{
+						Usages: []certificatesv1beta1.KeyUsage{
+							certificatesv1beta1.UsageKeyEncipherment,
+							certificatesv1beta1.UsageDigitalSignature,
+							certificatesv1beta1.UsageClientAuth,
+						},
+						Username: "system:serviceaccount:openshift-machine-config-operator:node-bootstrapper",
+						Groups: []string{
+							"system:authenticated",
+							"system:serviceaccounts:openshift-machine-config-operator",
+							"system:serviceaccounts",
+						},
+					},
+				},
+				csr: clientWithDNS,
+			},
+			wantErr: "Doesn't match expected prefix",
+		},
+		{
+			name: "client good but extra usage",
+			args: args{
+				machines: []machinev1beta1.Machine{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"machine.openshift.io/cluster-api-machine-role": "worker",
+								"machine.openshift.io/cluster-api-machine-type": "worker",
+							},
+						},
+						Status: machinev1beta1.MachineStatus{
+							Addresses: []corev1.NodeAddress{
+								{
+									Type:    corev1.NodeInternalDNS,
+									Address: "panda",
+								},
+							},
+						},
+					},
+				},
+				nodeName: "panda",
+				nodeErr:  errors.NewNotFound(schema.GroupResource{}, ""),
+				req: &certificatesv1beta1.CertificateSigningRequest{
+					Spec: certificatesv1beta1.CertificateSigningRequestSpec{
+						Usages: []certificatesv1beta1.KeyUsage{
+							certificatesv1beta1.UsageKeyEncipherment,
+							certificatesv1beta1.UsageDigitalSignature,
+							certificatesv1beta1.UsageClientAuth,
+							certificatesv1beta1.UsageServerAuth,
```

Comment:
We should have variants that have less usages and wrong ones (that replace client with server) for good measure.